### PR TITLE
STLX-793 from implementation transaction builder

### PR DESCRIPTION
**Pull request by smaria95**
Jira Ticket: [STLX-793](https://bitgoinc.atlassian.net/browse/STLX-793)

### CODE REVIEW

File: modules/account-lib/src/coin/cspr/transactionBuilderFactory.ts [hunk 1]
```diff
@@ -25,10 +25,17 @@ export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
...
-  from(raw: DeployUtil.Deploy): TransactionBuilder {
+  from(raw: string): TransactionBuilder {
     this.validateRawTransaction(raw);
     const tx = new Transaction(this._coinConfig);
-    tx.casperTx = raw;
+    const deployJson = JSON.parse(raw);
+    const deployObject = DeployUtil.deployFromJson(deployJson);
+
+    if (!deployObject) {
+      throw new InvalidTransactionError('Invalid transaction: ' + deployObject);
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transactionBuilderFactory.ts:35]: Can you create a const INVALID_TRANSACTION const

File: modules/account-lib/test/unit/coin/cspr/transactionBuilder/transactionBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import should from 'should';
+import { register } from '../../../../../src/index';
```
Comment (CapnMigraine) [inline modules/account-lib/test/unit/coin/cspr/transactionBuilder/transactionBuilder.ts:2]: nits:
```suggestion
import * as should from 'should';
import { register } from '../../../../../src';
```

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 3]
```diff
@@ -163,11 +169,15 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
 
   /** @inheritdoc */
-  validateRawTransaction(rawTransaction: any): void {
-    throw new NotImplementedError('validateRawTransaction not implemented');
-    // if (!isValidRawTransactionFormat(rawTransaction)) {
-    //   throw new ParseTransactionError('Invalid raw transaction');
-    // }
+  validateRawTransaction(rawTransaction: string): void {
+    if (!rawTransaction) {
+      throw new InvalidTransactionError('Raw transaction is empty');
```
[the remark this answers is not in the document]
Comment (smaria95) [reply modules/account-lib/src/coin/cspr/transactionBuilder.ts:174]: This is not required on source files.
We only use const for errors in test files.

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 4]
```diff
@@ -163,11 +169,15 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
+    }
+    try {
+      DeployUtil.deployFromJson(JSON.parse(rawTransaction));
+    } catch (e) {
+      throw new ParseTransactionError('There was an error parsing the JSON string');
```
[the remark this answers is not in the document]
Comment (smaria95) [reply modules/account-lib/src/coin/cspr/transactionBuilder.ts:179]: This is not required on source files.
We only use const for errors in test files.

File: modules/account-lib/src/coin/cspr/transactionBuilderFactory.ts (hunk 1 above)
Comment (smaria95) [reply modules/account-lib/src/coin/cspr/transactionBuilderFactory.ts:35]: This is not required on source files.
We only use const for errors in test files.

**Review by CapnMigraine — CHANGES_REQUESTED** [review]
Can you add the logic to disable the master key on wallet initialization?
The rest of the logic lgtm.

File: modules/account-lib/src/coin/cspr/walletInitializationBuilder.ts [hunk 5]
```diff
@@ -35,6 +35,12 @@ export class WalletInitializationBuilder extends TransactionBuilder {
     const accounts = this._owners.map(owner => CLTypedAndToBytesHelper.bytes(owner.address.toAccountHash()));
     const weights = this._owners.map(owner => CLTypedAndToBytesHelper.u8(owner.weight));
 
+    // set source address weight to zero to disable the master private key from signing.
+    accounts.push(
+      CLTypedAndToBytesHelper.bytes(PublicKey.fromHex(SECP256K1_PREFIX + this._source.address).toAccountHash()),
+    );
+    weights.push(CLTypedAndToBytesHelper.u8(0));
+
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/cspr/walletInitializationBuilder.ts:43]: 👍

**Review by CapnMigraine — APPROVED** [review]
lgtm

**Pull request merged**